src/DynamoCore/Graph/Workspaces/PackageDependencyInfo.cs

```diff
@@ -3,6 +3,18 @@
 
 namespace Dynamo.Graph.Workspaces
 {
+    /// <summary>
+    /// Enum containing the different types of package dependency states.
+    /// </summary>
+    internal enum PackageDependencyState
+    {
+        Loaded,
+        IncorrectVersion,
+        Missing,
+        Warning,
+        RequiresRestart
```

Comment:
Where is this set and how?

Reply:
This state is not yet assigned. This would happen when the user clicks on the "Keep local" button on the UI. Thats when it will trigger the uninstall of some package and this should be set. 

@QilongTang @mjkkirschner  Since the dependency table doesn't get refreshed until we reopen the graph or or click on the menu item, I was wondering how we are going to update the information in the extension tab. I think before handling this, it would be better to have more API's like closing the tab and refreshing the tab. What do you think?

Reply:
I am not too sure about the refresh part, but Close() and SwitchTo() will be useful for sure. Refresh() will be highly depending on if the extension author implemented this method

Reply:
Just saw this, can we call the DependencyRegen function that is in WorkspaceDependencyView.xaml.cs to generate the table again when user action is triggered.

Reply:
@reddyashish I think you mean if "Keep Local" is not chosen, in that case the state will be set to "RequiresRestart" in order to uninstall the local package and install the new one, isn't that right?

Reply:
@aparajit-pratap , yes sorry, that's right.